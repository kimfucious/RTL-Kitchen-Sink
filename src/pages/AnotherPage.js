import React from "react";
import { useHistory } from "react-router-dom";

export const AnotherPage = () => {
  const history = useHistory();
  return (
    <div
      className="d-flex flex-column justify-content-center align-items-center text-center w-100 full-height-with-nav text-light animate__animated animate__fadeIn"
      data-testid="another-page"
    >
      <div className="display-3 mb-3">
        <span role="img" aria-label="Cherry Blossom">
          ðŸŒ¸
        </span>
      </div>
      <p className="lead">This is another page</p>
      <button
        className="btn btn-link text-primary mt-2"
        onClick={() => history.push("/")}
      >
        back
      </button>
    </div>
  );
};
