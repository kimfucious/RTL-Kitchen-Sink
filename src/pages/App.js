import React, { useState } from "react";
import $ from "jquery";
import { useSelector, useDispatch } from "react-redux";
import { reverseUsername, getRandomUserData } from "../actions.js/userActions";
import { Spinner } from "../components/shared/Spinner";
import { UserDataModal } from "../components/UserDataModal";
import { getTitleCase } from "../helpers";

export const App = () => {
  const dispatch = useDispatch();
  const {
    user,
    user: { username }
  } = useSelector((state) => state);

  const [animation, setAnimation] = useState(false);
  const [error, setError] = useState("");
  const [isSpinning, setIsSpinning] = useState(false);

  const handleReverse = async () => {
    try {
      setError("");
      setIsSpinning("reversing");
      setAnimation("");
      await dispatch(
        reverseUsername(user.username, setIsSpinning, setAnimation)
      );
      setIsSpinning(false);
      setAnimation("animate__animated animate__tada");
    } catch (error) {
      setIsSpinning(false);
      setError(error.message);
    }
  };

  const handleGetUserData = async () => {
    try {
      setError("");
      setIsSpinning("fetching");
      await dispatch(getRandomUserData());
      setIsSpinning(false);
      $("#userDataModal").modal("show");
    } catch (error) {
      setIsSpinning(false);
      setError(error.message);
    }
  };

  return (
    <>
      <UserDataModal />
      <div
        className="d-flex flex-column align-items-center justify-content-center w-100 full-height-with-nav animate__animated animate__fadeIn"
        data-testid="app-page"
      >
        <span className="display-1" role="img" aria-label="desert island">
          üèùÔ∏è
        </span>
        <div
          className={`d-flex justify-content-center display-3 text-center text-muted mt-3`}
        >
          <span>Aloha, </span>
          <span className={`d-flex ml-3 ${animation}`} data-testid="username">
            {getTitleCase(username)}
          </span>
          !
        </div>
        <div className="d-flex align-items-center justify-content-center mt-5">
          <button
            className="d-flex align-items-center justify-content-center btn btn-outline-primary mr-2"
            disabled={isSpinning === "reversing"}
            onClick={() => handleReverse()}
            style={{ height: 38, width: 128 }}
          >
            {isSpinning === "reversing" ? (
              <div>
                <Spinner size={10} color="primary" type="beat" />
              </div>
            ) : (
              <span>Reverse Name</span>
            )}
          </button>
          <button
            className="d-flex align-items-center justify-content-center btn btn-outline-secondary ml-2"
            disabled={isSpinning === "fetching"}
            onClick={() => handleGetUserData()}
            style={{ height: 38, width: 128 }}
          >
            {isSpinning === "fetching" ? (
              <div>
                <Spinner size={10} color="secondary" type="beat" />
              </div>
            ) : (
              <span>Get New User</span>
            )}
          </button>
        </div>
        <div
          className={`mt-3 animate__animated animate__${
            error ? "fadeIn" : "fadeOut"
          }`}
        >
          <p className="text-warning">{error}</p>
        </div>
      </div>
    </>
  );
};
