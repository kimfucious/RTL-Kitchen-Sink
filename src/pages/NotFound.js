import React from "react";
import { useHistory } from "react-router-dom";

export const NotFound = () => {
  const history = useHistory();
  return (
    <div
      className="d-flex flex-column justify-content-center align-items-center text-center w-100 full-height-with-nav text-light animate__animated animate__fadeIn"
      data-testid="not-found-page"
    >
      <div className="display-3 mb-3">
        <span role="img" aria-label="Ghost emoji">
          ðŸ‘»
        </span>
      </div>
      <p className="lead">You still haven't found what you're looking for...</p>
      <button
        className="btn btn-link text-primary mt-2"
        onClick={() => history.replace("/")}
      >
        back
      </button>
    </div>
  );
};
