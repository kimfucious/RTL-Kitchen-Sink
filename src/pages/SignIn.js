import React, { useState } from "react";
import sink from "../images/sink.png";
import { useDispatch } from "react-redux";
import { Spinner } from "../components/shared/Spinner";
import { signIn } from "../actions.js/authActions";
import { setUser } from "../actions.js";

export const SignIn = () => {
  const dispatch = useDispatch();
  const [username, setUsername] = useState("");
  const [error, setError] = useState("");
  const [fade, setFade] = useState("");
  const [isSpinning, setIsSpinning] = useState("");

  const handleChange = (name) => {
    setFade("animate__animated animate__fadeOut");
    setUsername(name);
  };

  const handleSignin = async (e) => {
    try {
      e.preventDefault();
      setFade("animate__animated animate__fadeOut");
      setIsSpinning(true);
      if (!username) {
        throw new Error("Please enter a username");
      }
      if (username.length < 2) {
        throw new Error("Two characters minimum, please");
      }
      if (username.includes(" ")) {
        throw new Error("No spaces, please");
      }
      await dispatch(signIn(username.trim()));
      await dispatch(setUser(username.trim()));
    } catch (error) {
      setIsSpinning(false);
      setFade("animate__animated animate__fadeIn");
      setError(error.message);
    }
  };

  return (
    <div
      className="container d-flex flex-column align-items-center justify-content-center w-100 vh-100 text-muted animate__animated animate__fadeIn"
      data-testid="sign-in-page"
    >
      <div className="d-flex align-items-center justify-content-center display-3 mb-3">
        <span role="img" aria-label="RTL Octopus Logo">
          🐙
        </span>
        +
        <span>
          <img
            className="ml-2"
            src={sink}
            alt="sink"
            style={{ height: 84, width: 84 }}
          />
        </span>
      </div>
      <form
        className="d-flex flex-column align-items-center "
        data-testid="sign-in-form"
        onSubmit={handleSignin}
      >
        <div className="form-group mt-3">
          <label className="text-primary" htmlFor="usernameInput">
            Username
          </label>

          <input
            id="usernameInput"
            type="text"
            className="form-control myInput"
            placeholder="George"
            onChange={(e) => handleChange(e.target.value)}
            value={username}
          />
        </div>
        <button
          className="d-flex align-items-center justify-content-center btn btn-outline-primary mt-3"
          disabled={isSpinning}
          type="submit"
          style={{ height: 38, width: 76 }}
        >
          {isSpinning ? (
            <div>
              <Spinner size={10} color="primary" type="beat" />
            </div>
          ) : (
            <span>Sign In</span>
          )}
        </button>
      </form>
      <div className={`mt-3 ${fade}`}>
        <p className={`text-warning`}>{error}</p>
      </div>
    </div>
  );
};
