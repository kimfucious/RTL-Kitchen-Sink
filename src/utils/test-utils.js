import React from "react";
import { AppRouter } from "../routers/AppRouter";
import { Provider } from "react-redux";
import { render as rtlRender } from "@testing-library/react";
import configureStore from "../store/configureStore";

const render = (
  ui,
  {
    initialState = { auth: { userId: "", user: { username: "" } } },
    store = configureStore(initialState),
    ...renderOptions
  } = {}
) => {
  const Wrapper = ({ children }) => {
    return (
      <Provider store={store}>
        <AppRouter>{children}</AppRouter>
      </Provider>
    );
  };
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
};

export * from "@testing-library/react";

export { render };
