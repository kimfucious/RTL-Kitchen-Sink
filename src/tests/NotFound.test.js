import React from "react";
import { render, cleanup, fireEvent, waitFor } from "../utils/test-utils";
import { NotFound } from "../pages/NotFound";
import { createMemoryHistory } from "history";
import { App } from "../pages/App";

// const handleSignIn = jest.fn();

afterEach(cleanup);

describe("<NotFound/>", () => {
  test("renders", () => {
    const { asFragment } = render(<NotFound />);
    expect(asFragment()).toMatchSnapshot();
  });

  test.skip("renders", () => {
    const history = createMemoryHistory();
    history.push("/abc");
    const { getByTestId } = render(<App />);
  });
});
