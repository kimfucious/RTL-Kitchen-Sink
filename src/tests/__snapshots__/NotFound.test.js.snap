// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound/> renders 1`] = `
<DocumentFragment>
  <div
    class="container d-flex flex-column align-items-center justify-content-center w-100 vh-100 text-muted animate__animated animate__fadeIn"
    data-testid="sign-in-page"
  >
    <div
      class="d-flex align-items-center justify-content-center display-3 mb-3"
    >
      <span
        aria-label="RTL Octopus Logo"
        role="img"
      >
        üêô
      </span>
      +
      <span>
        <img
          alt="sink"
          class="ml-2"
          src="sink.png"
          style="height: 96px; width: 96px;"
        />
      </span>
    </div>
    <form
      class="d-flex flex-column align-items-center "
      data-testid="sign-in-form"
    >
      <div
        class="form-group mt-3"
      >
        <label
          class="text-primary"
          for="usernameInput"
        >
          Username
        </label>
        <input
          class="form-control myInput"
          id="usernameInput"
          placeholder="George"
          type="text"
          value=""
        />
      </div>
      <button
        class="d-flex align-items-center justify-content-center btn btn-outline-primary mt-3"
        style="height: 38px; width: 76px;"
        type="submit"
      >
        <span>
          Sign In
        </span>
      </button>
    </form>
    <div
      class="mt-3 "
    >
      <p
        class="text-warning"
      />
    </div>
  </div>
</DocumentFragment>
`;
